webhooks-extension/endpoints/credentials_test.go

```diff
@@ -0,0 +1,243 @@
+
+	// Verify no credentials have been created
+	checkCredentials(namespace, []credential{}, "", r, t)
+}
+
+func TestAccessTokenWithSecret(t *testing.T) {
+	r := dummyResource()
+	namespace := "default"
+	r.K8sClient.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+	accessTokenWithSecret := credential{
+		Name:        "accesstoken-with-secret",
+		AccessToken: "alongstringofcharacters",
+		SecretToken: "thisIsMySecretToken",
+	}
+	createAndCheckCredential(namespace, accessTokenWithSecret, "", r, t)
+}
+
+func TestAccessTokenWithNoSecret(t *testing.T) {
+	r := dummyResource()
+	namespace := "b-namespace"
+	r.K8sClient.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+
+	accessTokenNoSecret := credential{
+		Name:        "accesstoken-nosecret",
+		AccessToken: "anotherlongstringofcharacters",
+	}
+
+	jsonBody, _ := json.Marshal(accessTokenNoSecret)
+	t.Logf("json body for create cred test: %s", jsonBody)
+	httpReq := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/credentials?namespace=b-namespace", bytes.NewBuffer(jsonBody))
+	req := dummyRestfulRequest(httpReq, namespace, "")
+	httpWriter := httptest.NewRecorder()
+	resp := dummyRestfulResponse(httpWriter)
+	r.createCredential(req, resp)
+
+	httpReq = dummyHTTPRequest("GET", "http://wwww.dummy.com:8383/credentials?namespace=b-namespace", bytes.NewBuffer(nil))
+	req = dummyRestfulRequest(httpReq, namespace, "")
+	httpWriter = httptest.NewRecorder()
+	resp = dummyRestfulResponse(httpWriter)
+	r.getAllCredentials(req, resp)
+
+	result := []credential{}
+	b, err := ioutil.ReadAll(httpWriter.Body)
+	if err != nil {
+		t.Fatalf("FAIL: ERROR - reading response body: %s", err.Error())
+	}
+	err = json.Unmarshal(b, &result)
+
+	t.Logf("unmarshalled result '%+v'", result)
+
+	if result[0].Name != accessTokenNoSecret.Name {
+		t.Fatalf("Result came back with name %s but expected %s", result[0].Name, accessTokenNoSecret.Name)
+	}
+	// Finally check that result has a SecretToken set
+	if strings.Count(result[0].SecretToken, "") < 20 {
+		t.Fatalf("Result came back with less than twenty chars of secret token: '%s'", result[0].SecretToken)
+	}
+
+}
+
+func TestDeleteCredential(t *testing.T) {
+	r := dummyResource()
+	namespace := "ns2"
+	r.K8sClient.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+	accessTokenToDelete := credential{
+		Name:        "accesstokenToDelete",
+		AccessToken: "sdkfhighregusfbliusbbbwhfwiehw8hwefhw938hf497fhw97b47",
+	}
+	createAndCheckCredential(namespace, accessTokenToDelete, "", r, t)
+
+	httpReq := dummyHTTPRequest("GET", "http://wwww.dummy.com:8383/credentials/accesstokenToDelete?namespace=ns2", bytes.NewBuffer(nil))
+	req := dummyRestfulRequest(httpReq, namespace, "")
+	httpWriter := httptest.NewRecorder()
+	resp := dummyRestfulResponse(httpWriter)
+	r.deleteCredential(req, resp)
```

Comment:
do you want to add in a get and check the secret is no longer listed?

Reply:
doh! Done.